code/game/objects/effects/mines.dm

```diff
@@ -7,19 +81,42 @@
 	icon_state = "uglymine"
 	/// We manually check to see if we've been triggered in case multiple atoms cross us in the time between the mine being triggered and it actually deleting, to avoid a race condition with multiple detonations
 	var/triggered = FALSE
+	var/smart_mine = FALSE
+	var/disarm_time = 20 SECONDS
+	var/disarm_product = /obj/item/deployablemine // The item dropped on disarm.
+
+/obj/effect/mine/attackby(obj/I, mob/user, params)
+	if(istype(I, /obj/item/multitool))
+		to_chat(user, "<span class='notice'>You begin to disarm the [src]...</span>")
+		if(do_after(user, disarm_time, target = src))
+			to_chat(user, "<span class='notice'>You disarm the [src].</span>")
+			new disarm_product(src.loc)
+			qdel(src)
+			return
+	return ..()
 
 /obj/effect/mine/proc/mineEffect(mob/victim)
 	to_chat(victim, "<span class='danger'>*click*</span>")
 
-/obj/effect/mine/Crossed(atom/movable/AM)
-	if(triggered || !isturf(loc))
+/obj/effect/mine/Crossed(atom/movable/AM as mob|obj)
+	if(!isturf(loc) || AM.throwing || (AM.movement_type & (FLYING | FLOATING)) || !AM.has_gravity())
 		return
 	. = ..()
 
-	if(AM.movement_type & FLYING)
-		return
-
-	triggermine(AM)
+	if(ismob(AM))
+		checksmartmine(AM)
+	else
+		triggermine(AM)
```

Comment:
This logic is a bit convoluted. It's checking if it's a mob, then going to `checksmartmine()`
, then checking if it-s a `smart_mine` inside there, and then going back to `triggermine() anyway.
Chiefly, `checksmartmine()` is a no-operation if there's a `target` (which there should always be).

What you want is adding checks in `triggermine()` instead.

Reply:
Return could also work in that scenario, couldn't it?

Reply:
It should work